File: LeetCode/LeetCode Practice Problems/832- Flipping an Image.cpp
class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& A) {
        if(A.empty())
            return A;
        
        int n = A.size();
        int m = A[0].size();
        vector<vector<int>> s(n,vector<int>(m,0));
        
        for(int i=0;i<n;i++)
        {
            int k = m-1;
            for(int j=0;j<m;j++,k--)
                s[i][k] = A[i][j];
        }
        
        for(int i=0;i<n;i++)
            for(int j=0;j<m;j++)
                s[i][j] = abs(1-s[i][j]);
        
        return s;
    }
};

File: LeetCode/LeetCode Practice Problems/832-Flipping an Image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        int n = image.length;
        
        for(int i=0;i<n;i++) {
            int c1 = 0 , c2 = n-1;
            
            while(c1 <= c2) {
                if(c1 == c2) {
                    image[i][c1] ^= 1;
                } else {
                    int u = image[i][c1];
                    int v = image[i][c2];
                    
                    image[i][c1] = v ^ 1;
                    image[i][c2] = u ^ 1;
                }
                c1++;
                c2--;
            }
        }
        
        return image;
    }
}
